paddle/phi/kernels/gpu/multi_tensor_adam_kernel.cu

```diff
@@ -0,0 +1,278 @@
+                              const T* beta2_pow_,
+                              T* beta1_pow_out,
+                              T* beta2_pow_out) {
+  *beta1_pow_out = beta1 * beta1_pow_[0];
+  *beta2_pow_out = beta2 * beta2_pow_[0];
+}
+
+template <typename T, typename Context>
+void MultiTensorAdamKernel(
+    const Context& dev_ctx,
+    const std::vector<const DenseTensor*>& params,
+    const std::vector<const DenseTensor*>& grads,
+    const std::vector<const DenseTensor*>& moments1,
+    const std::vector<const DenseTensor*>& moments2,
+    const paddle::optional<std::vector<const DenseTensor*>>& master_param,
+    const DenseTensor& beta1_pow,
+    const DenseTensor& beta2_pow,
+    const DenseTensor& learning_rate,
+    const paddle::optional<DenseTensor>& skip_update,
+    const Scalar& beta1,
+    const Scalar& beta2,
+    const Scalar& epsilon,
+    int compute_group_size,
+    float weight_decay,
+    bool mode,
+    bool multi_precision,
+    bool use_global_beta_pow,
+    std::vector<DenseTensor*> params_out,
+    std::vector<DenseTensor*> moments1_out,
+    std::vector<DenseTensor*> moments2_out,
+    std::vector<DenseTensor*> master_param_out,
+    DenseTensor* beta1_pow_out,
+    DenseTensor* beta2_pow_out) {
+  using MPDType = typename phi::dtype::MPTypeTrait<T>::Type;
+
+  VLOG(4) << "use_global_beta_pow:" << use_global_beta_pow;
+  MPDType beta1_ = beta1.to<MPDType>();
+  MPDType beta2_ = beta2.to<MPDType>();
+  MPDType weight_decay_ = static_cast<MPDType>(weight_decay);
+  MPDType epsilon_ = epsilon.to<MPDType>();
+
+  bool skip_update_ = false;
+  if (skip_update.is_initialized()) {
+    PADDLE_ENFORCE_EQ(
+        skip_update->numel(),
+        1,
+        errors::InvalidArgument("Input(SkipUpdate) size must be 1, but get %d",
+                                skip_update->numel()));
+    std::vector<bool> skip_update_vec;
+    paddle::framework::TensorToVector(*skip_update, dev_ctx, &skip_update_vec);
+    skip_update_ = skip_update_vec[0];
+  }
+
+  // skip_update=true
+  // mutable_data
+  if (skip_update_) {
+    VLOG(4) << "Adam skip update";
+    for (int i = 0; i < params.size(); i++) {
+      phi::Copy(dev_ctx, *params[i], dev_ctx.GetPlace(), false, params_out[i]);
+      phi::Copy(
+          dev_ctx, *moments1[i], dev_ctx.GetPlace(), false, moments1_out[i]);
+      phi::Copy(
+          dev_ctx, *moments2[i], dev_ctx.GetPlace(), false, moments2_out[i]);
+    }
+    phi::Copy(dev_ctx, beta1_pow, beta1_pow.place(), false, beta1_pow_out);
+    phi::Copy(dev_ctx, beta2_pow, beta2_pow.place(), false, beta2_pow_out);
+    return;
+  }
+
+  std::vector<std::vector<DenseTensor*>> tensor_and_block_inf;
+
+  tensor_and_block_inf.push_back(params_out);
+  tensor_and_block_inf.push_back(moments1_out);
+  tensor_and_block_inf.push_back(moments2_out);
+  if (multi_precision) {
+    tensor_and_block_inf.push_back(master_param_out);
+  }
+
+  if (multi_precision) {
+    multi_tensor_adam_utility<5, MPDType>(dev_ctx,
```

Comment:
Please fix the function name:
 
- The `multi_tensor_adam_utility` is not like a function name, because it is a noun. Try to use another function name.
- Please use the Pascal case instead of camel case. Please follow the Google coding style https://google.github.io/styleguide/cppguide.html#General_Naming_Rules .